modules/social_features/social_follow_taxonomy/social_follow_taxonomy.module

```diff
@@ -467,3 +476,73 @@ function social_follow_tag_activity_send_email_notifications_alter(array &$items
     $items['what_follow']['templates'][] = 'update_node_following_tag';
   }
 }
+
+/**
+ * Implements hook_node_insert().
+ */
+function social_follow_taxonomy_node_insert(NodeInterface $node) {
+  _social_follow_taxonomy_invalidate_node_tagging_cache($node);
+}
+
+/**
+ * Implements hook_node_update().
+ */
+function social_follow_taxonomy_node_update(NodeInterface $node) {
+  _social_follow_taxonomy_invalidate_node_tagging_cache($node);
+}
+
+/**
+ * Implements hook_node_delete().
+ */
+function social_follow_taxonomy_node_delete(NodeInterface $node) {
+  _social_follow_taxonomy_invalidate_node_tagging_cache($node);
+}
+
+/**
+ * Invalidates cache for added/removed tags to node.
+ *
+ * @param \Drupal\node\NodeInterface $node
+ *   The node entity.
+ */
+function _social_follow_taxonomy_invalidate_node_tagging_cache(NodeInterface $node) {
+  // Only for nodes that has the 'social_tagging' field.
+  if (!$node->hasField('social_tagging')) {
+    return;
+  }
+
+  if ($node->isNew()) {
```

Comment:
@dmytriikaiun could you explain this code a little bit more?
Wondering if it will not have a performance impact since we are checking om each node change?
Also see `social_follow_tag_entity_update()` - can we combine it somehow, or it's not really related?

Reply:
@ribel I guess `social_follow_tag entity update` will not enough because it will just cover the case when we edit the existing node but also we should clear cache when adding a new node with tags to increase following items in follow term block and decrease when deleting a node with tags.

Reply:
Hi guys, good to see this discussion going on in here!
Did you already investigate if we can use a custom submit for this? This is way less impactful, and either adding, editing, or deleting a tag will be dealt with when passing this along correctly.
Also don't forget, we allow tags on entities (so more than just a node) like a group or a profile even I believe, so going for the node route isn't the best.

Reply:
@ronaldtebrake for now on the `Follow tag` block counter shows only related nodes and `See related content` redirects to the content search page. To add groups, profiles, etc. we need to change the counter logic, redirect users to the search all page, and add tag filters on it. If as `custom submit` you mean that we need to add it when the form has the `social_tagging` field it will not work on delete.

Reply:
Hi @dmytriikaiun it's not just about the block isn't it? 🤔 

![Screenshot 2021-02-01 at 11 08 55](https://user-images.githubusercontent.com/16667281/106444436-0e175880-647e-11eb-8a41-ac50ac075ebd.png)

Don't we want to invalidate caches for social_tagging regardless of this block? 

I'm ok with the block only being available on nodes, but as you can see in the screenshot we already add this as a field on a group, and the popup also works. Which redirects you nicely to the group search, with the social_tagging filter. But only after cache clear somethings happen. Which to me sounds like this isn't really done correctly.

Reply:
@ronaldtebrake I've added cache tags for the `Group tag` block. Also, fix the counter because even for groups it calculates related nodes.